### ISSUE TITLE
Subsetting during fitNULLGLMM

### ISSUE THREAD

**Issue by apoliakov** [weizhouUMICH/SAIGE#119]
Hello again!

This is a more of a question - something we'd like to confirm.

Suppose we've built a step 1 plink file X with a population P (1000 samples). Then suppose we'd like to do a SAIGE run (starting with fitNULLGLMM) with a subpopulation of, say 500 samples - a subset of P. Can we still use the file X?

In simpler terms, can then plink file input to fitNULLGLMM contain more samples than the phenotype data?

Based on all our experiments, looks like the answer is "yes" but just wanted to confirm with you.

Thanks a lot!

**Comment by weizhouUMICH** [comment]
Hi, 
SAIGE does not allow for non samples with missing dosages in step 2 but were used in step 1, mainly because the variance component parameter estimates from step 1 would be different if different samples are used in step 1. 

Thanks!
Wei

**Comment by apoliakov** [comment]
Hi Wei - thanks but this wasn't the question.

I get the Step 1 / Step 2 concordance. The question was about Step 1 and the plink file used for it.
In our workflows we create these plink files before step 1 (kind of a step 0 for us)

Can the plink file have more samples than used in step 1?

**Comment by weizhouUMICH** [comment]
Oh, I see. Sure. Plink file can definitely have more samples than used in step 1. 
Samples with both non-missing genotypes (in plink file) and non-missing phenotypes will be used for analysis in step 1.

**Comment by apoliakov** [comment]
Solid - thank you! This helps for sure.

**Comment by apoliakov** [comment]
Hi @weizhouUMICH !

I wanted to ask another question along this thread. Something that I'm confused about. Consider two cases.

### Case 1:
We run step 1 using 1000 people and then we run step 2 using genotypes for 500 people.

### Case 2:
We run step 1 using 1000 people, then we run step 2 using genotypes for 1000 people, but some variants have 500 missing genotypes.

Based on what you said above it looks like "Case 1" is not supported. But it also looks like SAIGE supports "Case 2" right now - it returns a result. Could you describe how these two are different? 

Thank you!

**Comment by weizhouUMICH** [comment]
Hi Alex,

In case 1, the reason why this scenario is not allowed is because the model
parameters estimated in step 1 would have different estimates if sample
sets are different. Theoretically, we should use consistent sample sets in
step 1 an step 2.

In case 2, presumably, there is only a small proportion of markers having
missing dosages/genotypes in a small subset of samples. For those markers
with missing dosages/genotypes, if the --IsDropMissingDosages=TRUE, samples
with missing dosages/genotypes will be dropped off in the tests for those
markers (although the assoc results won't be perfectly accurate with the
reason described above), otherwise, missing dosages/genotypes will be mean
imputed.

Thanks,
Wei

**Comment by apoliakov** [comment]
Thanks! I see.

So is it fair to say that if we are testing some variant with large missingness (like 50%) then we should really re-run step 1 on that subpopulation for better precision?

**Comment by weizhouUMICH** [comment]
Exactly.
